README.rst

```diff
@@ -13,16 +26,179 @@ Install the module with: ``pip install httpretty_fixtures``
+            res = requests.get('http://localhost:9200/my_index/my_document/my_id')
+
+fixture_manager.start(fixtures)
+"""""""""""""""""""""""""""""""
+Start running HTTPretty with a set of fixtures
+
+- fixtures ``list`` - Names of fixture functions to run
+
+  - \* ``str`` - Name of fixtures function to run
+
+
+This will run HTTPretty indefinitely until ``.stop()`` is run
+
+fixture_manager.stop()
+""""""""""""""""""""""
+Stop a running instance of HTTPretty. This should always be run at some point after a ``.start()``
+
+httpretty_fixtures.{verb}(\*register_uri_args, \*\*register_uri_kwargs)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Decorator to register a fixture function under an HTTP verb
+
+This is a summary for all possible HTTP verbs:
+
+.. code:: python
+    @httpretty_fixtures.get()
+    @httpretty_fixtures.put()
+    @httpretty_fixtures.post()
+    @httpretty_fixtures.delete()
+    @httpretty_fixtures.head()
+    @httpretty_fixtures.patch()
+    @httpretty_fixtures.options()
+    @httpretty_fixtures.connect()
+
+Each of these verbs functions passes its arguments/keyword arguments to ``HTTPretty's register_uri` function``.
+
+If there are any arguments you want to apply to your fixture with respect to ``HTTPretty``, this is how to do it.
+
+https://github.com/gabrielfalcao/HTTPretty
+
+httpretty_fixtures.first_request()
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Alias to access the first request received by ``HTTPretty``.
+
+**Warning:** If you are using ``HTTPretty`` in other locations, then this will register those requests as well.
+
+httpretty_fixtures.last_request()
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Alias to access the last request received by ``HTTPretty``.
+
+**Warning:** If you are using ``HTTPretty`` in other locations, then this will register those requests as well.
+
+httpretty_fixtures.requests()
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Alias to access all request received by ``HTTPretty``.
+
+**Warning:** If you are using ``HTTPretty`` in other locations, then this will register those requests as well.
 
 Examples
 --------
-_(Coming soon)_
+Preserving state between requests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+In this example, we will count between multiple requests to indicate that state is being preserved.
+
+.. code:: python
+
+    # Load in our dependencies
+    import unittest
+
+    import httpretty_fixtures
+    import requests
+
+
+    # Set up our fixture manager
+    class CounterServer(httpretty_fixtures.FixtureManager):
+        def __init__(self):
+            self.count = 0
+            super(CounterServer, self).__init__()
+
+        @httpretty_fixtures.get('http://localhost:9000/')
+        def counter(self, request, uri, res_headers):
+            self.count += 1
+            return (200, res_headers, str(self.count))
+
+
+    # Define our tests
+    class MyTestCase(unittest.TestCase):
+        @CounterServer.run(['counter'])
+        def test_counter_state(self):
```

Comment:
for this example and the test case below as well, it might be worth adding a second test case to show that the manager state is only maintained on a per test (start/stop) basis.


```python
@CounterServer.run(['counter'])
def test_counter_state(self):
    # make request
    self.asserEqual(res.text, '1')

@CounterServer.run(['counter'])
def test_counter_state_again(self):
    # make request
    self.assertEqual(res.text, '1')
```